src/Lib/Util/Password.hs

```diff
@@ -1,30 +1,45 @@
 {-# LANGUAGE FlexibleContexts #-}
 
+{-# LANGUAGE DeriveGeneric    #-}
+
 module Lib.Util.Password
-       ( PasswordHash
-       , PasswordPlainText
+       ( PasswordHash (..)
+       , PasswordPlainText (..)
        , mkPasswordHash
        , verifyPassword
        ) where
 
 import Control.Monad.Except (MonadError)
+import Data.Aeson (FromJSON (..), ToJSON (..), withText)
 
 import Lib.App.Error (AppError (..))
 import Lib.Util.App (maybeWithM)
 
 import qualified Crypto.BCrypt as BC
 
-type PasswordHash = Text
-type PasswordPlainText = Text
+newtype PasswordHash = PwdHash { unPwdHash :: ByteString }
+  deriving (Generic)
+
+instance ToJSON PasswordHash where
+  toJSON = toJSON . decodeUtf8 @Text . unPwdHash
```

Comment:
Well, strictly speaking, it's better to use [`decodeUtf8Strict`](https://github.com/serokell/universum/blob/a481e6076f94b2a1b6d67504f0bd17bc7bd17500/src/Universum/String/Conversion.hs#L91) function here because `decodeUtf8` is partial function. But, unfortunately, we can use safer function because `toJSON` is a pure function and should always return `Value`. We can patch `parseJSON` function to work with `Either` though as well.